src/sentry/api/endpoints/integrations/sentry_apps/details.py

```diff
@@ -82,7 +87,32 @@ def put(self, request: Request, sentry_app) -> Response:
 
     def delete(self, request: Request, sentry_app) -> Response:
         if sentry_app.is_unpublished or sentry_app.is_internal:
-            Destroyer.run(user=request.user, sentry_app=sentry_app, request=request)
+            if not sentry_app.is_internal:
+                for install in sentry_app.installations.all():
```

Comment:
I wonder how big this `all()` is in practice. It is theoretically unbounded, and could timeout. Perhaps we should be doing these deletes async in the future?

Reply:
Absolutely!  However, this matches the current behavior.  I went back and forth and decided to extract sentry app deletion into a separate PR because to do async also requires a status value and changes to the frontend.  I do intend to follow up.